lib/grpnice.gd

```diff
@@ -214,151 +337,121 @@ DeclareOperation(
-        function( obj1, obj2 )
-            if not IsIdenticalObj( NiceMonomorphism(obj1),
-                                NiceMonomorphism(obj2) )
-            then
-                TryNextMethod();
-            fi;
-            return oper( NiceObject(obj1), NiceObject(obj2) );
-        end );
+BindGlobal( "AttributeMethodByNiceMonomorphismCollColl", function( oper, par )
+ InstallNiceMonomorphismMethod( oper, par, "attribute CollColl",
+   IsIdenticalObj, [ 1, 2 ],
+    function( obj1, obj2 )
+      if not IsIdenticalObj( NiceMonomorphism(obj1),
+                             NiceMonomorphism(obj2) ) then
+        TryNextMethod();
+      fi;
+      return oper( NiceObject(obj1), NiceObject(obj2) );
+    end );
 end );
 
 
 #############################################################################
 ##
 #F  AttributeMethodByNiceMonomorphismCollElm( <oper>, <par> )
 ##
-##  <ManSection>
-##  <Func Name="AttributeMethodByNiceMonomorphismCollElm" Arg='oper, par'/>
-##
-##  <Description>
-##  </Description>
-##  </ManSection>
-##
 BindGlobal( "AttributeMethodByNiceMonomorphismCollElm", function( oper, par )
-
-    # check the argument length
-    if 2 <> Length(par)  then
-        Error( "need two arguments for ", NameFunction(oper) );
-    fi;
-    par    := ShallowCopy(par);
-    par[1] := par[1] and IsHandledByNiceMonomorphism;
-
-    # install the method
-    InstallOtherMethod( oper,
-        "handled by nice monomorphism: attribute CollElm",
-        IsCollsElms,
-        par,
-        0,
-        function( obj1, obj2 )
-            local   nice,img;
-            nice:=NiceMonomorphism(obj1);
-            img := ImagesRepresentative( nice, obj2:actioncanfail:=true );
-            if img = fail or
-              not (img in ImagesSource(nice) and
-                PreImagesRepresentative(nice,img)=obj2) then
-                TryNextMethod();
-            fi;
-            return oper( NiceObject(obj1), img );
-        end );
+  InstallNiceMonomorphismMethod( oper, par, "attribute CollElm",
+    IsCollsElms, [ 1 ],
+    function( obj1, obj2 )
+      local nice, img;
+      nice:= NiceMonomorphism( obj1 );
+      img := ImagesRepresentative( nice, obj2 : actioncanfail:= true );
+      if img = fail or
+         not ( img in ImagesSource( nice ) and
+               PreImagesRepresentative( nice, img ) = obj2 ) then
+        TryNextMethod();
+      fi;
+      return oper( NiceObject( obj1 ), img );
+    end );
 end );
 
+
 #############################################################################
 ##
-#F  AttributeMethodByNiceMonomorphismElmColl( <oper>, <par> )
-##
-##  <ManSection>
-##  <Func Name="AttributeMethodByNiceMonomorphismElmColl" Arg='oper, par'/>
-##
-##  <Description>
-##  </Description>
-##  </ManSection>
-##
-BindGlobal( "AttributeMethodByNiceMonomorphismElmColl", function( oper, par )
+#F  AttributeMethodByNiceMonomorphismCollElmOther( <oper>, <par>[, <meth>] )
+##
+BindGlobal( "AttributeMethodByNiceMonomorphismCollElmOther",
```

Comment:
This is new (just trying to make sense of this PR; enablingthe GitHub diff view option "hide whitespaces" also helps)

Reply:
Yes, some `AttributeMethodByNiceMonomorphism...` functions are new, analogous to the existing ones.
`AttributeMethodByNiceMonomorphismCollElmOther` is used in four places (in `lib/pcgsnice.gi`), I think this justifies the introduction of a function, compared to four individual methods.

And yes, some of the diffs shown are misleading.